bin/plot_truncated_iterations

```diff
@@ -0,0 +1,336 @@
+# copyright rights in the Software, and any and all patent rights owned or
+# freely licensable by each licensor hereunder covering either (i) the
+# unmodified Software as contributed to or provided by such licensor, or (ii)
+# the Larger Works (as defined below), to deal in both
+#
+# (a) the Software, and
+# (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+# one is included with the Software (each a "Larger Work" to which the Software
+# is contributed by such licensors),
+#
+# without restriction, including without limitation the rights to copy, create
+# derivative works of, display, perform, and distribute the Software and make,
+# use, sell, offer for sale, import, export, have made, and have sold the
+# Software and the Larger Work(s), and to sublicense the foregoing rights on
+# either these or other terms.
+#
+# This license is subject to the following condition: The above copyright
+# notice and either this complete permission notice or at a minimum a reference
+# to the UPL must be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+import copy
+import glob
+import json
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as pyplot
+from matplotlib.backends.backend_pdf import PdfPages
+import os
+import os.path
+import sys
+
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from warmup.krun_results import parse_krun_file_with_changepoints
+from warmup.plotting import add_margin_to_axes, compute_grid_offsets, style_axis, STYLE_DICT
+from warmup.summary_statistics import collect_summary_statistics
+
+# Set matplotlib styles, similar to Seaborn 'whitegrid'.
+for style in STYLE_DICT:
+    matplotlib.rcParams[style] = STYLE_DICT[style]
+
+# Default (PDF) font sizes
+TICK_FONTSIZE = 8
+AXIS_FONTSIZE = 8
+
+GRID_MAJOR_X_DIVS = 10
+GRID_MAJOR_Y_DIVS = 10
+
+LINE_WIDTH = 0.4
+MARKER_SIZE = 1
+
+DEFAULT_ITERS = 2000  # Iterations in original data file.
+ITER_STEP = 10  # Truncated data files each differ by ITER_STEP iterations.
+
+DEFAULT_SUMMARY_FILE = 'truncated_summary.json'
+
+pyplot.figure(tight_layout=True)
+
+
+def do_iqrs_differ((x1, y1), (x2, y2)):
+    """Given two IQRs return True if they do NOT overlap."""
+
+    assert (y1 > x1 or x1 == y1) and (y2 > x2 or x2 == y2)
+    return y1 < x2 or y2 < x1
+
+
+def do_cis_differ(mean1, ci1, mean2, ci2):
+    """Given two CIs return True if they do NOT overlap."""
+
```

Comment:
How can they be negative? If that does happen, it's probably a bug, and we'd probably end up printing CIs like "+/- -x" which would be... weird. So I'd suggest adding an `assert` that the CIs are `>= 0` and then simplify the code.